### PR TITLE
Changed trust region settings. Correct implementation.

### PR DESCRIPTION
**Pull request by mirraaj**
Paper implementation of trust region updates.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -57,45 +57,22 @@ def _update_networks(args, T, model, shared_model, shared_average_model, loss, o
...
-def _trust_region_loss(model, distribution, ref_distribution, loss, threshold):
-  _isolate_policy_grads(model, True)  # Disable gradients for other parameters
-  # Compute gradients from original loss
-  model.zero_grad()
-  loss.backward(retain_graph=True)
-  # Gradients should be treated as constants (not using detach as volatility can creep in when double backprop is not implemented)
-  g = [model.fc_actor.weight.grad.data.clone(), model.fc_actor.bias.grad.data]
-  model.zero_grad()
-
-  # KL divergence k ← ∇θ0∙DKL[π(∙|s_i; θ_a) || π(∙|s_i; θ)]
```
Comment (Kaixhin) [inline train.py:78]: Can we please keep this comment.

File: train.py [hunk 2]
```diff
@@ -57,45 +57,22 @@ def _update_networks(args, T, model, shared_model, shared_average_model, loss, o
...
-  kl = (ref_distribution * (ref_distribution.log() - distribution.log())).sum(1).mean(0)
-  # Compute gradients from (negative) KL loss (increases KL divergence)
-  (-kl).backward(retain_graph=True)
-  k = [model.fc_actor.weight.grad.data.clone(), model.fc_actor.bias.grad.data]
-  model.zero_grad()
+def _trust_region_loss(model, distribution, ref_distribution, loss, threshold, g, k):
+
+  kl = -(ref_distribution*(distribution.log()-ref_distribution.log())).sum(1).mean(0)
```
Comment (Kaixhin) [inline train.py]: Please put spaces between maths operators and variables (`*` and `-` here). Same on line 74.

**Review by Kaixhin — CHANGES_REQUESTED** [review]
Just a few minor comments, otherwise looks good! How does it perform?

**Comment by mirraaj** [comment]
This is the correct implementation according to me. The variance has reduced as discussed in the paper. I haven't played with hyperparameters yet. Just a random run gives this result(variance is evaluated every 5000 steps.)
newplot 6

I haven't used the --lr-decay. I think using that will make learning smoother.

**Review by Kaixhin — APPROVED** [review]

**Comment by Kaixhin** [comment]
Great, thank you very much.

**Pull request merged**